cf/dimensioncoordinate.py

```diff
@@ -334,6 +336,98 @@ def direction(self):
 
         return _direction
 
+    @classmethod
+    def createregular(
+        cls, range, interval, use_bounds=True, units=None, standard_name=None
+    ):
+        """
+        Create a new DimensionCoordinate with the given range and interval.
+
+        .. versionadded:: 3.15.0
+
+        :Parameters:
+
+            range: tuple
+                A tuple of two numeric values representing the lower
+            and upper bounds of the dimension.
+
+            interval: float
+                The interval between points along the dimension axis.
+
+            use_bounds: bool, optional
+                If True (default), the given range
+            represents the bounds, and the coordinate points will be the
+            midpoints of the bounds. If False, the given range represents the
+            coordinate points directly.
+
+            units: str, optional
+                The units of the DimensionCoordinate object.
+
+            standard_name: str, optional
+                The standard_name of the DimensionCoordinate object.
+
+        :Returns:
+
+            `DimensionCoordinate`
+                The newly created DimensionCoordinate object.
+
+        **Examples:**
+
+        >>> import cf
+        >>> longitude = cf.DimensionCoordinate.createregular((-180, 180), 1,
+            units='degrees_east', standard_name='longitude')
+        >>> longitude.dump()
+        Dimension coordinate: longitude
+            standard_name = 'longitude'
+            units = 'degrees_east'
+            Data(360) = [-179.5, ..., 179.5] degrees_east
+            Bounds:units = 'degrees_east'
+            Bounds:Data(360, 2) = [[-180.0, ..., 180.0]] degrees_east
+
+        """
+        if len(range) != 2:
+            raise ValueError("range must have two elements.")
+
+        range_diff = range[1] - range[0]
+        if range_diff <= 0:
+            raise ValueError("range must be positive.")
+
+        if range_diff % interval != 0:
+            raise ValueError(
+                "interval must be divisible by the range of the dimension."
+            )
+
+        if not isinstance(use_bounds, bool):
+            raise ValueError("use_bounds must be a boolean.")
+
+        if units is not None and not isinstance(units, str):
+            raise ValueError("units must be either None or a string.")
```

Comment:
Hi Ankit - this suggestion doesn't seem to have been applied - was that that intended?

Reply:
Hi David, you had suggested the following in https://github.com/NCAS-CMS/cf-python/pull/634#discussion_r1182885877:

> I realise that we don't need this, because the conversion to a `Units` object gets done for us inside the `Data` initialisation, below.